### PR TITLE
feat(configureStore): add support for `devTools` option accepting `EnhancerOptions`

### LINKED ISSUE #40: allow configuration of devtools via `configureStore`
There's only really one config value I care about and configure in my store for the devtools and that's the `name` but to support that and others, could the [`devtools`](https://github.com/reduxjs/redux-starter-kit#configurestore) option accept a boolean or a config object?

I'm opening an issue and not a PR because I'd like to see if the boolean or object is a decent solution or if another would be preferred. I can PR this though. 😄

### PR DESCRIPTION
**Pull request by wldcordeiro**
This is adding the code support for passing in `EnhancerOptions` from `redux-devtools-extension` for further customization.

https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#windowdevtoolsextensionconfig

fix #40

### CODE REVIEW

**Comment by wldcordeiro** [comment]
Going through and adding doc changes next, just wanted to get CI running to see how the code is looking.

File: package.json [hunk 1]
```diff
@@ -9,28 +9,27 @@
...
+    "@babel/preset-env": "^7.4.3",
+    "@babel/preset-typescript": "^7.3.3",
+    "@types/jest": "^23.3.14",
+    "@types/node": "^10.14.4",
     "@types/redux-immutable-state-invariant": "^2.1.0",
-    "babel-core": "7.0.0-bridge.0",
+    "@typescript-eslint/parser": "^1.6.0",
     "babel-eslint": "^10.0.1",
-    "eslint": "^4.17.0",
+    "eslint": "^5.16.0",
```
Comment (wldcordeiro) [inline package.json]: The major bump was safe for consumers of the CLI, only API users really needed to worry.

File: package.json [hunk 2]
```diff
@@ -49,12 +48,12 @@
     "src"
   ],
   "dependencies": {
-    "immer": "^1.10.5",
+    "immer": "^1.12.1",
     "redux": "^4.0.0",
-    "redux-devtools-extension": "^2.13.7",
+    "redux-devtools-extension": "^2.13.8",
```
Comment (wldcordeiro) [inline package.json]: Updating this was the main one to remove the `ts-ignore` but since I was here 🤷‍♂️

File: src/configureStore.ts [hunk 3]
```diff
@@ -133,12 +135,8 @@ export function configureStore<S = any, A extends Action = AnyAction>(
...
     finalCompose = composeWithDevTools({
       // Enable capture of stack traces for dispatched Redux actions
-
-      // @ts-ignore redux-devtools-extension doesn't have `trace` defined in
-      // its type definition file yet:
-      //
-      // https://github.com/zalmoxisus/redux-devtools-extension/pull/624
-      trace: !IS_PRODUCTION
+      trace: !IS_PRODUCTION,
+      ...typeof devTools === 'object' && devTools,
```
Comment (wldcordeiro) [inline src/configureStore.ts]: Perhaps I can do this in a different way? I played with a couple different ways, this just seemed terse and retained behavior.

File: package.json [hunk 4]
```diff
@@ -49,12 +49,12 @@
     "src"
   ],
   "dependencies": {
-    "immer": "^1.10.5",
+    "immer": "^2.1.5",
```
Comment (wldcordeiro) [inline package.json]: This one may be no? Immer now resolves Promises returned from produce.

Comment (markerikson) [reply package.json]: Yeah, let's stay on 1.x for the time being.  I know they've been making some changes over there, and I'm not clear what all is different.  (For example, there was discussion of ignoring values returned from `produce()` in favor of just draft mutations, but I don't know how that turned out.)

Comment (wldcordeiro) [reply package.json]: Cool cool, also saw they're like in the middle of the WIP 3.x release over there too 😆

Comment (wldcordeiro) [reply package.json]: I will update to latest 1.x

File: docs/api/configureStore.md [hunk 5]
```diff
@@ -52,7 +52,9 @@ For more details on how the `middleware` parameter works and the list of middlew
 
 ### `devTools`
 
-A boolean indicating whether `configureStore` should automatically enable support for [the Redux DevTools browser extension](https://github.com/zalmoxisus/redux-devtools-extension).
+If this is a boolean, it will be used to indicate  whether `configureStore` should automatically enable support for [the Redux DevTools browser extension](https://github.com/zalmoxisus/redux-devtools-extension).
+
+If it is an object of `redux-devtools-extension` [`EnhancerOptions`](https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#windowdevtoolsextensionconfig), `configureStore` will use the user provided configuration options.
```
Comment (wldcordeiro) [inline docs/api/configureStore.md]: I kept your note about the `trace` option below but this option description maybe read a little awkwardly. I wasn't sure how to best update.

**Comment by wldcordeiro** [comment]
@markerikson ready for review here, pushed up the last of my changes.

File: package.json (hunk 4 above)
Comment (markerikson) [reply package.json]: So having said that, someone else just filed an "Update to Immer 2.x" PR and I merged it, so let's go with that.

**Comment by markerikson** [comment]
Awright, rebased against master after merging a couple other PRs, and looks good.  Thanks!

**Pull request merged**